src/check/parse/tokenize.zig

```diff
@@ -637,6 +670,49 @@ pub const Cursor = struct {
         var sawNewline = false;
         var indent: u16 = 0;
 
+        // Configurable SIMD fast path: process chunks of spaces and tabs
+        if (comptime build_options.simd_enabled) {
+            const SIMD_WIDTH = build_options.simd_width;
+            const VecType = @Vector(SIMD_WIDTH, u8);
+
+            while (self.pos + SIMD_WIDTH <= self.buf.len) {
+                const chunk: VecType = self.buf[self.pos .. self.pos + SIMD_WIDTH][0..SIMD_WIDTH].*;
+
+                // Use SIMD builtins to check if all characters are either spaces or tabs
+                const spaces: VecType = @splat(' ');
+                const tabs: VecType = @splat('\t');
+                const is_space = chunk == spaces;
+                const is_tab = chunk == tabs;
+                const true_vec: @Vector(SIMD_WIDTH, bool) = @splat(true);
+                const is_trivial = @select(bool, is_space, true_vec, is_tab);
+                const all_trivial = @reduce(.And, is_trivial);
+
+                // If not all characters are spaces or tabs, fall back to scalar processing
+                if (!all_trivial) {
+                    break;
+                }
+
+                // Process the chunk - all characters are spaces or tabs
+                if (sawNewline) {
```

Comment:
Isn't `sawNewline` always `false` here and everywhere it is checked?

Reply:
Good pickup. I missed that. I'm currently trying to get a realistic input to test it on and verify it is actually processing SIMD chunks.